p2pserver/message/utils/msg_workerpool.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2018 The ontology Authors
+ * This file is part of The ontology library.
+ *
+ * The ontology is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ontology is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with The ontology.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package utils
+
+import (
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/ontio/ontology-eventbus/actor"
+	"github.com/ontio/ontology/common/log"
+	"github.com/ontio/ontology/p2pserver/message/types"
+	"github.com/ontio/ontology/p2pserver/net/protocol"
+)
+
+type msgJobChanList *[]*msgJobChan
+
+type msgWorkerPool struct {
```

Comment:
this worker pool implementation looks familiar to me. If I remember correctly, it is copied and modified from `fasthttp`, so add original copyright in this file.

Reply:
Yes, the worker pool is modified based on fasthttp. I have added copyright of fasthttp.